src/gt4py/next/common.py

```diff
@@ -623,14 +623,17 @@ def asnumpy(self) -> np.ndarray: ...
     def remap(self, index_field: ConnectivityField | fbuiltins.FieldOffset) -> Field: ...
 
     @abc.abstractmethod
-    def restrict(self, item: AnyIndexSpec) -> Field | core_defs.ScalarT: ...
+    def restrict(self, item: AnyIndexSpec) -> Field: ...
+
+    @abc.abstractmethod
+    def as_scalar(self) -> core_defs.ScalarT: ...
```

Comment:
@egparedes Wondering if Field should not be a protocol (or we should separate an ABC), because then we could do the check for 0d field and put the concrete implementation in a separate function. but would become obsolete if we refactor to a single field implementation

Reply:
I don't completely follow because in any case a Protocol is an ABC, so you can put implementation details in the protocol for the case where other specific implementations want to use it as an ABC and inherit from it. With Protocols the inheritance is optional if implementations provide exactly the same interface.